src/main/java/com/google/firebase/database/core/RepoManager.java

```diff
@@ -134,6 +142,26 @@ public void run() {
     }
   }
 
+  private void destroyInternal(final Context ctx) {
```

Comment:
This seems to be a subset of interruptInternal() [it interrupts all the repos but never calls ctx.stop()].  Could we just use interruptInternal()?

Reply:
We were effectively calling `interruptInternal()` before this PR. But it only calls `ctx.stop()` conditionally. Specifically, it won't close the context when there are active event listeners. The new `destroy()` method has stronger semantics in the sense it will always close the context.

Reply:
Yeah.  My point was that destroy() could call interrupt() and then call ctx.stop() instead of duplicating this code.  I don't care that strongly though.